### DISCUSSION TITLE
Specifying a delimiter in broker transport options causes kombu to fail

### DISCUSSION THREAD

**Discussion by sondrelg** [celery/celery#7849] [General]
Seems a discussion should be had before filing a bug report, so opening this here :slightly_smiling_face: 

## Context

According to the documentation [here](https://docs.celeryq.dev/en/latest/userguide/routing.html#redis-message-priorities), you can use the `priority` queue ordering strategy, which will create queues looking like this:

```
['celery', 'celery\x06\x163', 'celery\x06\x166', 'celery\x06\x169']
```

If you prefer though, you can specify a delimiter. This configuration

```python
CELERY_BROKER_TRANSPORT_OPTIONS = {
    'queue_order_strategy': 'priority',
    'priority_steps': list(range(4)),
    'sep': '_',
}
```

Should generate queues with these names

```
['celery', 'celery', 'celery_1', 'celery_2', 'celery_3']
```

## The issue

Kombu has a reply.celery table in redis (`smembers _kombu.binding.reply.celery.pidbox` will retrieve its values). These values are normally interpreted as

```
[('3cc3df90-5124-3715-a9de-58607b491174', '', '3cc3df90-5124-3715-a9de-58607b491174.reply.celery.pidbox'),]
```

By kombu, but when specifying the `_` separator, it produced an entry in the table like this:

```
[("909a4346-8c15-3398-b375-cddd2e8cfea3__909a4346-8c15-3398-b375-cddd2e8cfea3.reply.celery.pidbox")]
```

Which lead to 

```
Traceback (most recent call last):
  File \"/opt/venv/lib/python3.10/site-packages/celery/worker/pidbox.py\", line 44, in on_message
    self.node.handle_message(body, message)
  File \"/opt/venv/lib/python3.10/site-packages/kombu/pidbox.py\", line 141, in handle_message
    return self.dispatch(**body)
  File \"/opt/venv/lib/python3.10/site-packages/kombu/pidbox.py\", line 108, in dispatch
    self.reply({self.hostname: reply},
  File \"/opt/venv/lib/python3.10/site-packages/kombu/pidbox.py\", line 145, in reply
    self.mailbox._publish_reply(data, exchange, routing_key, ticket,
  File \"/opt/venv/lib/python3.10/site-packages/kombu/pidbox.py\", line 275, in _publish_reply
    producer.publish(
  File \"/opt/venv/lib/python3.10/site-packages/kombu/messaging.py\", line 177, in publish
    return _publish(
  File \"/opt/venv/lib/python3.10/site-packages/kombu/connection.py\", line 523, in _ensured
    return fun(*args, **kwargs)
  File \"/opt/venv/lib/python3.10/site-packages/kombu/messaging.py\", line 199, in _publish
    return channel.basic_publish(
  File \"/opt/venv/lib/python3.10/site-packages/kombu/transport/virtual/base.py\", line 595, in basic_publish
    return self.typeof(exchange).deliver(
  File \"/opt/venv/lib/python3.10/site-packages/kombu/transport/virtual/exchange.py\", line 69, in deliver
    for queue in _lookup(exchange, routing_key):
  File \"/opt/venv/lib/python3.10/site-packages/kombu/transport/virtual/base.py\", line 704, in _lookup
    R = self.typeof(exchange).lookup(
  File \"/opt/venv/lib/python3.10/site-packages/kombu/transport/virtual/exchange.py\", line 61, in lookup
    return {
  File \"/opt/venv/lib/python3.10/site-packages/kombu/transport/virtual/exchange.py\", line 62, in <setcomp>
    queue for rkey, _, queue in table
ValueError: not enough values to unpack (expected 3, got 1)",
```

because `queue for rkey, _, queue in table` is called, where table is normally a tuple of these 3 items, and now it no longer is.

tl;dr: Seems like specifying a delimiter for redis priority-ordered queues breaks something upstream

Am I missing something obvious here, or does this seem like a legitimate bug? I don't have the time to create an mvp right now, but might be able to later if needed :bow:
